py/client/build.gradle

```diff
@@ -81,14 +81,11 @@ deephavenDocker {
     networkName.set "pydeephaven-network-${randomSuffix}"
 }
 
-def testPyClient = Docker.registerDockerTask(project, 'testPyClient') {
+def buildPyClientImage = Docker.registerDockerTask(project, 'pyClient') {
```

Comment:
The "main/test" split that you're doing shouldn't be necessary in this project - the py-client project already uses our home-grown "python-wheel" plugin, which will make a wheel file on disk ready to be used by the test. 

If you want to run the tests against the wheel, I'd suggest dropping this task here, and just copy the wheel into the test docker image. Otherwise it is appropriate to do the test against the sources instead of the already-installed wheel (but you might have reasons to not do that?).

Very roughly, entirely remove this task, and copy the already-built wheel and requirements-dev.txt, and pip install both. Here's an example of what testPyClient could look like (it runs and seems to pass locally) - I copied the dev dependencies first under the assumption that they change less frequently.

```
def testPyClient = Docker.registerDockerTask(project, 'testPyClient') {
    copyIn {
        from(tasks.named('buildWheel').get().outputs.files.singleFile)
        from ('requirements-dev.txt') {
            into 'project/'
        }
        from('tests') {
            into 'project/tests'
        }
    }
    containerDependencies.dependsOn = [deephavenDocker.healthyTask]
    containerDependencies.finalizedBy = deephavenDocker.endTask
    network = deephavenDocker.networkName.get()
    dockerfile {
        from('deephaven/cpp-client:local-build')
        copyFile('project/requirements-dev.txt', '/project/requirements-dev.txt')
        runCommand('pip3 install -r /project/requirements-dev.txt')
        copyFile('*.whl', '/')
        runCommand('pip3 install /pydeephaven*.whl')
        copyFile('project', '/project')
        workingDir('/project')
        environmentVariable 'DH_HOST', deephavenDocker.containerName.get()
        environmentVariable 'DH_PORT', '10000'
    }
    parentContainers = [ project.tasks.getByPath(':cpp-client:cppClient') ]
    entrypoint = ['python3', '-m', 'xmlrunner', 'discover', 'tests', '-v', '-o', '/out/report']
    copyOut {
        into layout.buildDirectory.dir('test-results')
    }
}
```

Reply:
Done.